art/classifiers/tensorflow.py

```diff
@@ -311,6 +315,10 @@ def loss_gradient(self, x, y, **kwargs):
         if not hasattr(self, "_loss_grads") or self._loss_grads is None or self._labels_ph is None:
             raise ValueError("Need the loss function and the labels placeholder to compute the loss gradient.")
 
+        # Check label shape
+        if len(self._labels_ph.shape) == 1:
```

Comment:
If following comment above, we could replace this with `self._reduce_labels`.

Reply:
Done.